core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalMaxTests.java

```diff
@@ -45,4 +55,31 @@ protected InternalMax createTestInstance(String name, List<PipelineAggregator> p
     protected void assertReduced(InternalMax reduced, List<InternalMax> inputs) {
         assertEquals(inputs.stream().mapToDouble(InternalMax::value).max().getAsDouble(), reduced.value(), 0);
     }
+
+    public void testFromXContent() throws IOException {
+        InternalMax max = createTestInstance();
+        boolean humanReadable = randomBoolean();
+        XContentType xContentType = randomFrom(XContentType.values());
+        ToXContent.Params params = new ToXContent.MapParams(Collections.singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
+        BytesReference originalBytes = toXContent(max, xContentType, params, humanReadable);
+
+        InternalMax parsed;
+        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+            assertEquals(max.getWriteableName() + "#" + max.getName(), parser.currentName());
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            parsed = InternalMax.parseXContentBody(max.getName(), parser);
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+        assertEquals(max.getName(), parsed.getName());
+        assertEquals(max.getValue(), parsed.getValue(), Double.MIN_VALUE);
+        // we don't write Double.NEGATIVE_INFINITY to xContent, so we cannot always recreate the exact String representation
```

Comment:
ah here it is... I guess it's just a matter of picking positive or negative and getting over the fact that we can't have both. null would be better here but we can't have it.